cmd/skupper/skupper.go

```diff
@@ -750,20 +759,26 @@ func NewCmdUnbind(newClient cobraFunc) *cobra.Command {
 	return cmd
 }
 
+func IsZero(v reflect.Value) bool {
+	return !v.IsValid() || reflect.DeepEqual(v.Interface(), reflect.Zero(v.Type()).Interface())
+}
+
 func NewCmdVersion(newClient cobraFunc) *cobra.Command {
-	// TODO: change to inspect
 	cmd := &cobra.Command{
 		Use:    "version",
 		Short:  "Report the version of the Skupper CLI and services",
 		Args:   cobra.NoArgs,
 		PreRun: newClient,
 		RunE: func(cmd *cobra.Command, args []string) error {
 			silenceCobra(cmd)
-
 			fmt.Printf("%-30s %s\n", "client version", client.Version)
-			fmt.Printf("%-30s %s\n", "transport version", cli.GetVersion(types.TransportComponentName, types.TransportContainerName))
-			fmt.Printf("%-30s %s\n", "controller version", cli.GetVersion(types.ControllerComponentName, types.ControllerContainerName))
-
+			if !IsZero(reflect.ValueOf(cli)) {
```

Comment:
can you not just test if cli != nil?

Reply:
I found that I could not as a cli != nil check would crash, iiuc related to the pointer versus underlying value, there may be a more direct way but found this approach as recommended